src/zjs_gpio.c

```diff
@@ -13,6 +15,73 @@
 #include "zjs_callbacks.h"
 #include "zjs_promise.h"
 
+#if defined(ZJS_LINUX_BUILD) || defined(QEMU_BUILD)
+#ifndef ZJS_GPIO_MOCK
+#define ZJS_GPIO_MOCK
```

Comment:
It would depend on the definition of "QEMU_BUILD" (there're different hw targets for QEMU for example), but I'd consider it too coarse a logic for "if it's qemu, use mock GPIO". A QEMU target may mock GPIO on its own for example.

Perhaps, use numeric values for ZJS_GPIO_MOCK, and #if (not #ifdef) to test it?